packages/libs/react-components/src/utils/convertStyles.ts

```diff
@@ -0,0 +1,27 @@
+import { Primitive } from '../typings';
+
+export const convertStyles = (
+  styles: Record<string, Primitive | Record<string, Primitive>>,
+): Record<string, Primitive | Record<string, Primitive>> => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const newStyles = {} as any;
+  Object.entries(styles).forEach(([key, value]) => {
+    if (value === undefined) return;
+
+    if (typeof value === 'object') {
+      return Object.keys(value).forEach((mediaQueryKey) => {
+        if (newStyles['@' + mediaQueryKey] === undefined) {
+          newStyles['@' + mediaQueryKey] = {};
+        }
+
+        const prop = styles[key];
+        if (typeof prop === 'object') {
+          newStyles['@' + mediaQueryKey][key] = prop[mediaQueryKey];
+        }
+      });
+    }
+    return (newStyles[key] = value);
+  });
+
+  return newStyles;
```

Comment:
I noticed this doesn't account for sizes smaller than what is defined. I think it would be beneficial to implement an `@initial` like stitches does so they can also set a default value
https://stitches.dev/docs/responsive-styles#setting-an-initial-variant

Reply:
shouldnt the initial be in the component it self already?